src/App.tsx

```diff
@@ -1,61 +1,114 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import React, { useState } from 'react';
+import { Todo } from './types/Todo';
+
+function findUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId);
+}
+
+const todos = todosFromServer.map(todo => {
+  const user = findUserById(todo.userId);
+
+  return { ...todo, user };
+});
 
 export const App = () => {
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [title, setTitle] = useState('');
+  const [updatedTodos, setUpdatedTodos] = useState<Todo[]>(todos);
+  const [titleError, setTitleError] = useState(false);
+  const [selectedUserError, setSelectedUserError] = useState(false);
+
+  // <Todo[]>
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+
+    if (!selectedUser || !title) {
+      if (!selectedUser) {
+        setSelectedUserError(true);
+      }
+
+      if (!title) {
+        setTitleError(true);
+      }
+
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: Math.max(...updatedTodos.map(todo => todo.id)) + 1,
+      title: title,
+      completed: false,
+      userId: selectedUser,
+      user: findUserById(selectedUser),
+    };
+
+    setUpdatedTodos([...updatedTodos, newTodo]);
+
+    setSelectedUser(0);
+    setTitle('');
+  }
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="todo-title">
+            {'Title: '}
+            <input
+              placeholder="Enter a title"
+              type="text"
+              data-cy="titleInput"
+              id="todo-title"
+              value={title}
+              onChange={event => {
```

Comment:
Would be better to ceparate this function, as it have more than one action

Reply:
Good point, fix this